File: math/LCM/C/LCM.c
#include <stdio.h> 

long long __gcd(long long a, long long b) {
	if(b == 0) 
		return a;
	return __gcd(b, a % b);
}

long long __lcm(long long a, long long b) {
	return a * b / __gcd(a, b);
}

int main() {
	long long num1, num2;
	printf("Please enter two numbers.\n");
	scanf("%lld %lld", &num1, &num2);

	long long lcm = __lcm(num1, num2);
	printf("LCM of %lld and %lld is %lld\n", num1, num2, lcm);
	return 0;
}

File: math/LCM/Python/LCM.py
#Program to find the LCM of two numbers

#Function to find GCD
def gcd(num1, num2):
    
    if num1 == num2:
        return num1
    
    if num1 > num2:
        return gcd(num1-num2, num2)
    
    return gcd(num1, num2-num1)

#Function to find LCM
def lcm(num1, num2):
    return (num1*num2) // gcd(num1, num2)

#Driver function for testing above
def test():
    num1, num2 = 12, 4
    print('LCM of {} and {} is {}'.format(num1, num2, lcm(num1, num2)))
    

    

